core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -687,6 +690,10 @@ private Expression translate0(RexNode expr, RexImpTable.NullAs nullAs,
       Expression input = list.append("inp" + index + "_", x); // safe to share
       return handleNullUnboxingIfNecessary(input, nullAs, storageType);
     }
+    case LAMBDA_REF: {
+      final String name = ((RexLambdaRef) expr).getName();
+      return Expressions.parameter(typeFactory.getJavaClass(expr.getType()), name);
```

Comment:
I remove this and the tests still passes, so, it seems not necessary.

Reply:
This is important and there are test failures `FAILURE  49.5sec,  277 completed,   1 failed,   1 skipped, org.apache.calcite.test.CalciteSqlOperatorTest
FAILURE  83.9sec, 5925 completed,   3 failed, 104 skipped, Gradle Test Run :core:test`